internal/pkg/manager/spod/spod_controller.go

```diff
@@ -426,11 +426,33 @@ func (r *ReconcileSPOd) getConfiguredSPOd(
 			templateSpec.Containers,
 			r.baseSPOd.Spec.Template.Spec.Containers[bindata.ContainerIDBpfRecorder])
 
-		// HostPID is only required for the bpf recorder and is used to access
+		// HostPID is required for the bpf recorder and is used to access
 		// cgroup files to map Process IDs to Pod IDs.
 		templateSpec.HostPID = true
 	}
 
+	// AppArmor parameters
+	if cfg.Spec.EnableAppArmor {
+		falsely, truly := false, true
+		var userRoot int64 = 0
+		// a more privileged mode is required when apparmor is enabled
+		// TODO: review security model and provide a dynamic approach that can be case specific
+		templateSpec.Containers[bindata.ContainerIDDaemon].SecurityContext = &corev1.SecurityContext{
+			AllowPrivilegeEscalation: &truly,
+			Privileged:               &truly,
+			ReadOnlyRootFilesystem:   &falsely,
+			RunAsUser:                &userRoot,
+			RunAsGroup:               &userRoot,
+		}
```

Comment:
@saschagrunert @jhrozek @JAORMX this ensures that spod is privileged/root only when apparmor is enabled - which would work as a feature switch and therefore cause no impact to existing execution levels for Seccomp/SELinux. PTAL

I will follow-up with some materials to foster discussion around current security model and a few ideas on changes we could adopt - but that will be a separate thread/PR.

Reply:
I think this is fine, but have you considered instead the model we have used for SELinux where the privileged operations are executed by a separate workload, selinuxd in this case?
But as I said, I'm OK with this because it doesn't increase the set of privileges for neither the default case or the case we'll be enabling for openshift, but only for this one case.
Thank you for thinking about security!

Reply:
Did we get to any consensus here? Is AppArmor gonna switch to a similar sidecar container model that the SELinux part uses? or will we take this as-is

Reply:
@jhrozek  @JAORMX we spoke a bit off-line on this on last catch-up. I am correct assuming we are in consensus on the approach: merge as-is and then propose a better apporach on follow-up PR. Would you be OK with that?